GoogleDriveAccessor.py

```diff
@@ -0,0 +1,198 @@
+from DataGenerator import DataGenerator
+
+from googleapiclient.http import MediaIoBaseDownload
+from googleapiclient.discovery import build
+from httplib2 import Http
+from oauth2client import file, client, tools
+from CSVReader import CSVReader
+
+import io
+import os
+import time
+import argparse
+from enum import Enum
+from multiprocessing import Lock
+
+SCOPES = 'https://www.googleapis.com/auth/drive'
+
+class FileState(Enum):
+    NOT_STARTED = 1
+    DOWNLOADING = 2
+    DOWNLOADED = 3
+
+class SetActions(Enum):
+    START_DOWNLOADING = 1
+    WAIT_FOR_DOWNLOAD = 2
+    START_PROCESSING = 3
+
+
+class GoogleDriveAccessor(DataGenerator):
+    def __init__(self, folder_id=None, output_folder="data", file_name=None, file_type="CSV"):
+        self.__folder_ID = folder_id
+        self.__output_folder = self.__process_path(path=output_folder)
+        self.__service = None
+        self.__file_name = file_name
+        self.__file_type = file_type
+        self.__file_download_status = None
+        self.__lock = Lock()
+        self.__data_loader = None
+
+    def __check_download_status(self):
+        if os.path.isfile(self.__output_folder + self.__file_name):
+            return FileState.DOWNLOADED
```

Comment:
It downloads it as a zip then uncompresses it at the end, so I would assume that until it finishes downloading to decompress it would not be the correct file.